_posts/2020-05-03-static-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+veth interface. A veth interface type comes always in pairs, anything that
+ingress one edge, egress the other edge and vice versa. It provides a L2 local
+connectivity between the peers.
+
+We will create the veth interface at the root namespace and then place each
+peer in one of the namespaces.
+
+Lets create the veth interface:
+```
+sudo ip link add veth00 type veth peer name veth10
+```
+And place each in the relevant namespace:
+```
+sudo ip link set dev veth00 netns red
+sudo ip link set dev veth10 netns blue
+```
+
+At this point, assuming that IPv6 is enabled on the host, each namespace
+should show an loopback interface and another veth type interface which
+has a [link-local IPv6 address](https://tools.ietf.org/html/rfc4291#page-11).
+
+To check the interfaces and their addresses, execute:
+```
+sudo ip netns exec red ip addr
+sudo ip netns exec blue ip addr
+```
+
+## L3 connectivity
+With the IPv6 link-local addresses in place, we can already check the
+connectivity betweem the two namespaces.
+
+Using the infromation gathered in the previous `ip addr` commands, learn
+the link-local IPv6 address of each peer and use it in the ping command.
+
+Note: As the addresses have link-local scope, a zone must be added to the
+destination address.
+A link-local address has a default subnet of 64 bits with a default network
+address of `fe80`. Therefore, on a node with multiple interfaces, an explicit
+egress interface needs to be provided in order for the packet to know through
+which interface to exit. 
+See [here](https://tools.ietf.org/html/rfc4007) for more information
+about zones.
+
+Lets run an IPv6 ping:
+(on older distributions use `ping -6` and on newer ones use `ping6`)
+```
+sudo ip netns exec red ping6 fe80::<the-peer-last-64-bits-address%veth00
+``` 
+
+## IPv6 with global scope connectivity
+The previous IPv6 link-local addresses may be used to check L3 connectivity
+betweem two directly connected peers (i.e. interfaces connected to the same
+physical LAN). Routers are required not to forward link-local addresses.
+
+Therefore, in order to enable IPv6 connectivity beyond the physical LAN,
+a global scoped address needs to be defined on the interface.
+Such addresses may be set staically or dynamically.
+
+### IPv6 Global Static Address
+In order to enable connectivity without a router, we need both peers to be
+set on the same network subnet, i.e. the network prefix need to be identical
+for both peers and the host part needs to be unique.
+
+We will use a 64 bit network subnet with a network address of `fd00`.
+Resulting in the following addresses:
+- red: `fd00::11/64`
+- blue: `fd00::22/64`
+
+Lets define a static address for each namespace:
+```
+sudo ip netns exec red ip addr add fd00::11/64 dev veth00
+sudo ip netns exec blue ip addr add fd00::22/64 dev veth10
+```
+With this behind us, we can check the connectivity (without the zone part
+this time):
+```
+sudo ip netns exec red ping -6 fd00::22
+```
+
```

Comment:
Consider adding a link to the following post.

Reply:
Done